### PR TITLE
ci: Mono repo setup

### PR DESCRIPTION
**Pull request by frascuchon**
The mono repo structure will divide the argilla repository into different folders, 
each one containing a different part of the project. 

The `.github/workflows` folder will contain the different workflows for each part of the project because of the
limitation of GitHub Actions to have only one workflow folder per repository.

The `docs` (or `argilla-docs`) folder will contain the documentation for the whole project.

The rest of the folders will contain the different parts of the project, such as the SDK, 
the frontend, and the server.

Each part of the project will have its own `CHANGELOG.md` file and its own `README.md` file.

Contribution guidelines will be in the `CONTRIBUTING.md` file and the code of conduct will be in the `CODE_OF_CONDUCT.md` file.

A example of the project structure would be:

```
- .github/workflows/
    |_ argilla.yml # test + build argilla
    |_ argilla-frontend.yml # test + build argilla-frontend
    |_ argilla-server.yml # test + build argilla-server
    ...
- argilla-sdk/ #  -> V2
    |_ CHANGELOG.md
    |_ README.md
    |_ src/
    |_ tests/
    ...
- argilla/ # -> V1
    |_ CHANGELOG.md
    |_ README.md
    |_ src/
    |_ tests/
    ...
- argilla-frontend/
    |_ CHANGELOG.md
    |_ README.md
    |_ package.json
    |_ nuxt.config.ts
	...
- argilla-server/
    |_ CHANGELOG.md
    |_ README.md
    |_ src/
    |_ tests/
    ...
- docs/
    ...
- CONTRIBUTING.md
- CODE_OF_CONDUCT.md
...
```

### CODE REVIEW

**Comment by frascuchon** [comment]
This PR is ready to merge once the incoming release is published.

File: .github/workflows/argilla-server.yml [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+    types:
+      - opened
+      - edited
+      - reopened
+      - synchronize
+      - ready_for_review
+
+  release:
+    types:
+      - "published"
```
Comment (jfcalvo) [inline .github/workflows/argilla-server.yml]:
```suggestion
      - published
```

File: .github/workflows/argilla-server.yml [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+        uses: pdm-project/setup-pdm@v4
+        with:
+          python-version-file: argilla-server/pyproject.toml
+          cache-dependency-path: argilla-server/pdm.lock
+          cache: true
+
+      - name: Install dependencies
+        run: pdm install
+
+      - name: Run tests 📈
```
Comment (jfcalvo) [inline .github/workflows/argilla-server.yml:76]: We're running tests only with PostgreSQL?

Comment (frascuchon) [reply .github/workflows/argilla-server.yml:76]: Yes. I will include a separate step to run tests with SQLite (and other python engines) later. 

I decided to keep this first version quite simple, without matrix configurations.

**Comment by jfcalvo** [comment]
@damianpumar @leiyre can you take a look to the commit  https://github.com/argilla-io/argilla/pull/4742/commits/7e7ff9653223d63bc0f26622054ac2696cb73d44 updating `argilla-frontend` folder with the latests changes from `develop` before merging the PR?

**Pull request merged**